src/main/java/io/github/amerebagatelle/fabricskyboxes/mixin/skybox/FogColorMixin.java

```diff
@@ -27,11 +29,11 @@ public class FogColorMixin {
      */
     @Inject(method = "render(Lnet/minecraft/client/render/Camera;FLnet/minecraft/client/world/ClientWorld;IF)V", at = @At(value = "FIELD", target = "Lnet/minecraft/client/render/BackgroundRenderer;lastWaterFogColorUpdateTime:J", ordinal = 6))
     private static void modifyColors(Camera camera, float tickDelta, ClientWorld world, int i, float f, CallbackInfo ci) {
-        if (SkyboxManager.shouldChangeFog) {
-            red = SkyboxManager.fogRed;
-            blue = SkyboxManager.fogBlue;
-            green = SkyboxManager.fogGreen;
-            SkyboxManager.shouldChangeFog = false;
+        Skybox skybox = SkyboxManager.getInstance().getCurrentSkybox();
+        if (skybox instanceof FSBSkybox fsbSkybox && fsbSkybox.getAlpha() > SkyboxManager.MINIMUM_ALPHA && fsbSkybox.getProperties().isChangeFog()) {
```

Comment:
Curious, when is the first one not true here?  Also, I'm confused why it would benefit to make the fog color change inaccessible to other mods.

Reply:
Mostly because `Skybox` interface only has a few generic methods for rendering and priority, other mods can use this to implement their own skybox type without using `FSBSkybox` implementation.